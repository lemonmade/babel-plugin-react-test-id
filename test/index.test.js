import * as babel from 'babel-core';
import babelPluginReactTestID from '../src';

describe('babelPluginReactTestID()', () => {
  it('removes a testID prop', () => {
    expect(transform(`
      const foo = <Foo testID="bar" />
    `)).toMatchSnapshot();

    expect(transform(`
      const foo = <Foo testID={someValue} />
    `)).toMatchSnapshot();
  });

  it('preserves other props', () => {
    expect(transform(`
      const foo = <Foo test="bar" />
    `)).toMatchSnapshot();
  });

  describe('props', () => {
    it('removes custom prop names', () => {
      expect(transform(`
        const foo = <Foo data-test-id="bar" />
      `, {props: ['data-test-id']})).toMatchSnapshot();
    });

    it('does not remove testID if custom names are provided', () => {
      expect(transform(`
        const foo = <Foo testID="bar" />
      `, {props: ['data-test-id']})).toMatchSnapshot();
    });
  });
});

function transform(code, options) {
  return babel.transform(code, {
    babelrc: false,
    plugins: [
      [babelPluginReactTestID, options],
    ],
    parserOpts: {
      plugins: ['jsx'],
    },
  }).code.trim();
}
