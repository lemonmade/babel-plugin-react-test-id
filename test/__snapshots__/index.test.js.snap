// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babelPluginReactTestID() preserves other props 1`] = `"const foo = <Foo test=\\"bar\\" />;"`;

exports[`babelPluginReactTestID() props does not remove testID if custom names are provided 1`] = `"const foo = <Foo testID=\\"bar\\" />;"`;

exports[`babelPluginReactTestID() props removes custom prop names 1`] = `"const foo = <Foo />;"`;

exports[`babelPluginReactTestID() removes a testID prop 1`] = `"const foo = <Foo />;"`;

exports[`babelPluginReactTestID() removes a testID prop 2`] = `"const foo = <Foo />;"`;
